### ISSUE TITLE
minig stops after ~24h

### ISSUE THREAD

**Issue by gurkburk76** [RainbowMiner/RainbowMiner#923]
So i have an amd rig with 4 570 cards what work just fine in windows but i wanted to switch to linux (less laggy if/when i actually need to get into t he computer to fix/change things etc) and that works well except for the fact that it seems to maybe crash something and stop mining after ~ 24h, i've seen this behaviour 3 days in a row so it would be nice if the culpit of the issue could be found.

I've pull out a debug file via the webinterface but i'm a bit reluctand to upload it to some random website .. can i get a hold of you on discord or other means ? (i've already tried on discord to discuss another issue but it seems like you don't hang there alot) :)

**Comment by RainbowMiner** [comment]
You can just upload it here, in this thread. The Debug file has all wallet addresses, keys and api secrets stripped, alreay. Alternatively, send it by eMail to rbminer2018@gmail.com

**Comment by gurkburk76** [comment]
Well ok if those things are stripped i guess it's ok ;)
I just started it up again and it actually crashed after ~15 mins after a few mins so something is up, i also check the temp of the cards and the hottest one goes to 74c so that should be within reason.

Running it again, we'll see how it behaves.

[debug_2020-03-21.zip](https://github.com/RainbowMiner/RainbowMiner/files/4363212/debug_2020-03-21.zip)

**Comment by RainbowMiner** [comment]
Very weird - there is nothing specific happening.
- Is the rig crashing completely, so that you need a reboot?
- did you run "install.sh" before starting RainbowMiner?
- also, please try to start "ocdaemon status" - is it running?
- is the current user "root"? Note: there is no need to start RainbowMiner as root, as long as the ocdaemon is up and running in the background ("ocdaemon status"). But if you did so, already, you will have to install RainbowMiner into a different folder, copy the config files and chown oder chmod them, so that a normal linux user can read/write them all.

One fact, maybe: it seems the mining crashes while running energy intense miners. Eventually, there is an overload on the PSU. But nonetheless it should never stop completely.

**Comment by gurkburk76** [comment]
1) no it just seems like that when a miner crashes it does not restart.
2) yes
3) it is, i'm however not sure if it's actually doing anything on a AMD rig (you don't seem to be able to controll things like powercapping on amp on linux, i think it's even mentioned in the docs)
4) yes i'm running it as root to avoid weird things.

the cards are drawing 450w at most (some miner software actually shows the power drawn even in linux) and the psu is around 1500w and it's not crashing when i run it in windows,, or if it is it's at least restarting.
Like i wrote before, in windows i haven't tried running rainbowminer for any extended period, but i've ran MultiPoolMiner in windows for months without issues.

**Comment by gurkburk76** [comment]
And there it crashed again, when mining for dev :)

Actually i think the last time before this crash it was *also* mining for dev... could it have something to do with the coin and/or pool it uses when it mines for you ?

**Comment by gurkburk76** [comment]
![bild](https://user-images.githubusercontent.com/37435605/77255786-4a790600-6c6a-11ea-9f7c-57f09b7d93d9.png)

**Comment by gurkburk76** [comment]
I'll switch to windows on monday just to see if it makes any diff with rbm on that rig.

**Comment by RainbowMiner** [comment]
Ah! That's a nice hint. I will check, what is going on.

**Comment by gurkburk76** [comment]
I have another small nvidia rig (x2 1060 3gig + 1 1070ti) on that same network running on windows that works flawlessly on rbm, could be good to know :)

**Comment by RainbowMiner** [comment]
I bet it is due to miners, that are incompatible with AMD on linux. I'll dig a bit deeper.

**Comment by gurkburk76** [comment]
I just thought i give you some more info, stopped again at this:

![bild](https://user-images.githubusercontent.com/37435605/77260050-d2b8d480-6c85-11ea-947d-7150aeb6af99.png)

**Comment by gurkburk76** [comment]
![bild](https://user-images.githubusercontent.com/37435605/77286773-44b80a80-6cd4-11ea-817e-56440d98d251.png)

**Comment by gurkburk76** [comment]
switched to windows OS, seems to be ok a day in.. too bad the miners on linux seem to behave erratic.
If you manage to solve it i'll switch back to linux and try it.
